### PR TITLE
Add get last method for HttpHeaders and QueryParams

### PR DESCRIPTION
**Pull request by kojilin**
Motivation: Close https://github.com/line/armeria/issues/3438

Modifications:
* Add getLastXxx for all getXxx

### CODE REVIEW

**Comment by kojilin** [comment]
Didn't change the impl to use cyclic, getLast is finding the 1st in the entries.

**Review by minwoox — APPROVED** [review]
👍 👍 👍

File: core/src/test/java/com/linecorp/armeria/common/HttpHeadersBaseTest.java [hunk 1]
```diff
@@ -70,6 +70,15 @@ void testEqualsInsertionOrderDifferentHeaderNames() {
         assertThat(h1).isEqualTo(h2);
     }
 
+    @Test
+    void testGetLastOperations() {
+        final HttpHeadersBase headers = newEmptyHeaders();
+        headers.add("Foo", "1");
+        headers.add("Foo", "2");
+
+        assertThat(headers.getLast("Foo")).isEqualTo("2");
```
Comment (ikhoon) [inline core/src/test/java/com/linecorp/armeria/common/HttpHeadersBaseTest.java:79]: Could you add test cases for a single value and an empty value? 😄

**Review by ikhoon — APPROVED** [review]
Thanks! @kojilin

**Review by trustin — APPROVED** [review]
Thanks for working on this, @kojilin! @anuraaga will be happy about this.
